# Getting familiar with Movies_Metadata File

Reload dataset

In [275]:
path="C:/Users/Elsa/OneDrive/master/03_reve/data/"
metadata_file="movies_metadata.csv"
metadata=pd.read_csv(path+metadata_file,low_memory=False)
reves_metadata=metadata[metadata["revenue"]>0]
print(reves_metadata.shape)
print(reves_metadata.columns)

(7408, 24)
Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')


#### Loading libraries and data

In [1]:
import pandas as pd
import numpy as np
import matplotlib as plt

In [2]:
#Set the path to the data files
path="C:/Users/Elsa/OneDrive/master/03_reve/data/"
metadata_file="movies_metadata.csv"

In [3]:
metadata=pd.read_csv(path+metadata_file,low_memory=False)
#The reason you get this low_memory warning is because guessing dtypes for each column is very memory demanding. 
#Pandas tries to determine what dtype to set by analyzing the data in each column

#### Analyzing what we have

In [4]:
print(type(metadata))
print(metadata.shape)
print(metadata.columns)

<class 'pandas.core.frame.DataFrame'>
(45466, 24)
Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')


We have a 45466 row DataFrame, with 24 columns or variables. At first sight, we are interested in the following columns
- Id
- Belongs to collection
- Budget
- Genres
- Popularity
- Production companies
- Production countries
- Release date
- **Revenue**
- Runtime
- Title


In [5]:
metadata.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 24 columns):
adult                    45466 non-null object
belongs_to_collection    4494 non-null object
budget                   45466 non-null object
genres                   45466 non-null object
homepage                 7782 non-null object
id                       45466 non-null object
imdb_id                  45449 non-null object
original_language        45455 non-null object
original_title           45466 non-null object
overview                 44512 non-null object
popularity               45461 non-null object
poster_path              45080 non-null object
production_companies     45463 non-null object
production_countries     45463 non-null object
release_date             45379 non-null object
revenue                  45460 non-null float64
runtime                  45203 non-null float64
spoken_languages         45460 non-null object
status                   45379 non-null objec

In [6]:
metadata.isnull().sum()

adult                        0
belongs_to_collection    40972
budget                       0
genres                       0
homepage                 37684
id                           0
imdb_id                     17
original_language           11
original_title               0
overview                   954
popularity                   5
poster_path                386
production_companies         3
production_countries         3
release_date                87
revenue                      6
runtime                    263
spoken_languages             6
status                      87
tagline                  25054
title                        6
video                        6
vote_average                 6
vote_count                   6
dtype: int64

In [7]:
#print (metadata[pd.to_numeric(metadata['id'], errors='coerce').isnull()])

Apparently, seems like there are no null, but let's take a deeper look at each column

In [8]:
metadata[metadata.columns].dtypes

adult                     object
belongs_to_collection     object
budget                    object
genres                    object
homepage                  object
id                        object
imdb_id                   object
original_language         object
original_title            object
overview                  object
popularity                object
poster_path               object
production_companies      object
production_countries      object
release_date              object
revenue                  float64
runtime                  float64
spoken_languages          object
status                    object
tagline                   object
title                     object
video                     object
vote_average             float64
vote_count               float64
dtype: object

- Budget should be numeric
- Belongs_to_collection should be bool
- Id should be numeric

#### 0. Id

This will be the Key to join the different dataframes analyzed in this project. As we have established before, there are no missing values for this key-column, but type is wrong because it shouldn't be object but numeric. 

In [9]:
metadata["id"] =pd.to_numeric(metadata['id'], errors='coerce',downcast="integer")

In [10]:
metadata["id"].dtype

dtype('float64')

#### 1. Revenue

In [11]:
metadata["revenue"].describe()

count    4.546000e+04
mean     1.120935e+07
std      6.433225e+07
min      0.000000e+00
25%      0.000000e+00
50%      0.000000e+00
75%      0.000000e+00
max      2.787965e+09
Name: revenue, dtype: float64

In [12]:
metadata["revenue"].isnull().sum()

6

The whole ponint to our model is being able to predict a movie revenue, so, this column is essential for this project. That's why the first thing we should do is find out how many movies do we have that actually have a registered value for the revenue. If this value is null, we will have to drop those rows.

As we found out in the previous cell, and also checked at metadata's stats, there are 6 null values in Revenue. Furthermore, we ought to check if we have values worth $0 revenue, because they'd be like null for us. 

In [13]:
metadata[metadata["revenue"]>0].shape

(7408, 24)

Out of 45466 movies, there are only 7408 rows with revenue metadata. We are going to create a new dataframe containing only the movies with revenue information, since it's essencial for our model. From now on, we would be using this new dataframe named "reves_metadata".

In [14]:
reves_metadata=metadata[metadata["revenue"]>0]

In [15]:
print(reves_metadata.shape)
print(reves_metadata.columns)

(7408, 24)
Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')


The new file has 7408 rows with the same 24 columns

In [16]:
print(reves_metadata["revenue"].describe())
print("-----")
print("There are %d null values in revenue column in this new dataframe" % (reves_metadata["revenue"].isnull().sum()))

count    7.408000e+03
mean     6.878739e+07
std      1.464203e+08
min      1.000000e+00
25%      2.400000e+06
50%      1.682272e+07
75%      6.722707e+07
max      2.787965e+09
Name: revenue, dtype: float64
-----
There are 0 null values in revenue column in this new dataframe


Now let's check some random values. They seem like reasonable numbers for a movie revenue, don't they?

In [17]:
reves_metadata["revenue"].sample(10)

10760     17496992.0
41049      4073489.0
38222         7636.0
39920    125729635.0
38189     14397593.0
13122     28814580.0
3101      29381956.0
43057     26800152.0
6628      10000000.0
3476       1568749.0
Name: revenue, dtype: float64

#### 2. Belongs to Collection

In [46]:
reves_metadata["belongs_to_collection"].head(10)

0     {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                   NaN
3                                                   NaN
4     {'id': 96871, 'name': 'Father of the Bride Col...
5                                                   NaN
8                                                   NaN
9     {'id': 645, 'name': 'James Bond Collection', '...
10                                                  NaN
12    {'id': 117693, 'name': 'Balto Collection', 'po...
13                                                  NaN
Name: belongs_to_collection, dtype: object

It seems that this column contains 2 different types of values: 

- If, indeed, the movie belongs to a series or a collection, the row contains a dict with the Id and the Name of that collection. 
- Otherwise, if the movie does not belong to a collection, the value is NaN. 

For our model, the valuable information is to know whether a movie belongs to a collection or not, because our assumption is that if it does, the expected revenue could be higher. 

Values different from NaN mean that they actually belong to a collection, so this could be simplified if translated into a dummie column where 0 means "Doesn't belong to collection" and 1 "Does belong to a collection". 

We will also create a  new column named "Collection info" with the id of the collection, in order to use it in the future if necessary. 

In [47]:
type(reves_metadata["belongs_to_collection"][0])

str

reves_metadata["belongs_to_collection"]=reves_metadata["belongs_to_collection"].replace(np.nan,"0")
reves_metadata["belongs_to_collection"].head(5)

In [48]:
reves_metadata["belongs_to_collection"][0]

"{'id': 10194, 'name': 'Toy Story Collection', 'poster_path': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg', 'backdrop_path': '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'}"

In [49]:
type(reves_metadata["belongs_to_collection"][0])

str

In [50]:
reves_metadata["belongs_to_collection"][1]

nan

In [51]:
type(reves_metadata["belongs_to_collection"][1])

float

In [58]:
for element in reves_metadata["belongs_to_collection"]:
    if type(element) == float:
        element=="0"
    elif type(element)== str:
        element=="1"
        #reves_metadata["belongs_to_collection"][element].replace(element,"1")

In [59]:
reves_metadata["belongs_to_collection"].head(5)

0    {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                  NaN
3                                                  NaN
4    {'id': 96871, 'name': 'Father of the Bride Col...
5                                                  NaN
Name: belongs_to_collection, dtype: object

In [30]:
type(reves_metadata["belongs_to_collection"][1])

float

We have different type of data in this Pandas Series:
- Dict values --> str
- NaN values --> float

In [15]:
reves_metadata["collection"]=reves_metadata["belongs_to_collection"].copy()

C:\Users\elsad\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [28]:
reves_metadata["belongs_to_collection"][1]

nan

In [ ]:
for index,value in reves_metadata["belongs_to_collection"].items():
    reves_metadata["belongs_to_collection"][index]=value
    if type (value) == str:
        value=1
        print(value)
    else:
        value=0
        print(value)
    

C:\Users\elsad\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  
C:\Users\elsad\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:2862: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  exec(code_obj, self.user_global_ns, self.user_ns)


1
0
0
1
0
0
1
0
1
0
0
0
0
0
1
0
1
0
0
0
0
0
0
1
0
0
1
0
0
0
1
0
0
0
0
1
1
1
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
1
1
0
1
0
0
0
0
1
0
1
0
0
0
0
0
0
0
0
0
1
0
1
0
0
1
0
1
1
0
1
0
0
0
0
0
1
0
0
1
0
1
0
0
1
0
0
1
0
0
0
1
0
0
0
0
0
0
1
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
1
0
0
0
1
0
0
0
0
0
0
0
1
0
0
1
0
1
0
1
0
0
0
0
0
0
1
0
1
1
1
0
0
1
1
0
0
0
0
0
0
1
1
0
1
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
1
0
0
0
1
1
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
1
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
1
1
0
1
1
0
1
0
0
0
0
1
0
1
1
0
0
0
1
0
0
0
0
1
1
0
0
1
1
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
1
0
0
0
0
0
1
1
0
1
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
1
0
1
0
0
0
1
1
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
1
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
1
0
0
0
1
0
0
0
0
0
0
1
0
1
1
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
1
1
0
0
0
0
0
0
0
0
0
0
0
0
1
0
0
0
1
1
0
0
0
0
0
0
0
0
0
0
0
0
0


In [37]:
reves_metadata["belongs_to_collection"].head()

0    {'id': 10194, 'name': 'Toy Story Collection', ...
1                                                    0
3                                                    0
4    {'id': 96871, 'name': 'Father of the Bride Col...
5                                                    0
Name: belongs_to_collection, dtype: object

In [ ]:
#hint! Use replace(what,with)

#### 3. Budget

First, basic data exploring

In [18]:
reves_metadata["budget"].head()

0    30000000
1    65000000
3    16000000
4           0
5    60000000
Name: budget, dtype: object

In [19]:
reves_metadata["budget"].describe()

count     7408
unique     705
top          0
freq      2027
Name: budget, dtype: object

Surprise! Seems like values are not numeric! We should fix that.

In [20]:
reves_metadata["budget"]=pd.to_numeric(reves_metadata["budget"], errors="coerce")
reves_metadata["budget"]=reves_metadata["budget"].replace(0,np.nan)

C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.
C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


In [21]:
reves_metadata["budget"].dtype

dtype('float64')

Now we have to analyze, out of the 7408, how many movies contain budget information in order to determine if we can use this variable in our model. As it happened with "Revenue" seems like we have not nulls, but we have false positives, since a 0 budget is null for us and a perfectly "not null" value for Python. This column should be numeric and $0 budget should be Nan.

In [22]:
reves_metadata[reves_metadata["budget"]>0]["budget"].count()

5381

In [23]:
reves_metadata["budget"].sample(5)

28533    38000000.0
4686     28000000.0
1713      1000000.0
4381     24000000.0
730      25000000.0
Name: budget, dtype: float64

Out of 7404 movies, 5381 have a positive value for budget. Is that enough data for our model?
Also, taking a deeper look at the values, seems that most of them are exact and round amounts, and they don't seem very accurate. Should we use this data in our model?

#### 4. Genres

In [24]:
reves_metadata["genres"].sample(5)

14822    [{'id': 35, 'name': 'Comedy'}, {'id': 10749, '...
2182                        [{'id': 35, 'name': 'Comedy'}]
3310     [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...
1858     [{'id': 12, 'name': 'Adventure'}, {'id': 10751...
3209                   [{'id': 99, 'name': 'Documentary'}]
Name: genres, dtype: object

In [25]:
print(reves_metadata["genres"][0])
print(reves_metadata["genres"][1])
print(reves_metadata["genres"][235])

[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]
[{'id': 12, 'name': 'Adventure'}, {'id': 14, 'name': 'Fantasy'}, {'id': 10751, 'name': 'Family'}]
[{'id': 10749, 'name': 'Romance'}, {'id': 16, 'name': 'Animation'}, {'id': 10751, 'name': 'Family'}, {'id': 35, 'name': 'Comedy'}, {'id': 12, 'name': 'Adventure'}]


Seems that in this column, we can find a list of dicts that indicate the different movie genres associated to each movie. As we can see, there are many different generes associated to each movie, but not always the same ammount of them. First of all, we should find out how many different genres we've got, in order to decide what to do and how to use this information. 

Basic questions we should ask ourselves about the column "Genres":
- Should we create dummie variables with all the genres? 
- What's the maxium numbers of genres per movie we want to use in our model 
- ...
- Maybe we shoul just keep the id_genre info, and save the dict of (index,genre) outside the main dataframe, just for querys.


First, let's find out how many different genres we have got, and what's the maximum number of genres any of the movies has

In [26]:
#We will create a function as we will use it later with some other columns
def to_list (column):
    from ast import literal_eval
    new_column= column.fillna('[]').apply(literal_eval).apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])
    return new_column

In [27]:
reves_metadata["genres"]=to_list(reves_metadata["genres"])

C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [257]:
reves_metadata["genres"].head(10)

0          [Animation, Comedy, Family]
1         [Adventure, Fantasy, Family]
3             [Comedy, Drama, Romance]
4                             [Comedy]
5     [Action, Crime, Drama, Thriller]
8        [Action, Adventure, Thriller]
9        [Adventure, Action, Thriller]
10            [Comedy, Drama, Romance]
12      [Family, Animation, Adventure]
13                    [History, Drama]
Name: genres, dtype: object

In [28]:
genres_lenghts=[]
for line in reves_metadata["genres"]:
    lenght=len(line)
    genres_lenghts.append(lenght)
print("There are a max of %d genres per movie"%(max(genres_lenghts)))

There are a max of 8 genres per movie


How many genres per movie are we using for out model?

In [29]:
np.mean(genres_lenghts)

2.4966252699784017

Maybe we should set up the maximux on 3. It's a reasonable number of categories for a variable. In case we do that, we would be assuming that the first 3 genres mentioned in this column would be the most reliable, and the ones that best describe the movie's storyline.

Let's see how many genres we've got just in case we want to create dummie coluns for each of them.  

In [30]:
#We will create a function as we are going to do the same process with Production Companies and Production Countries.

def get_uniques(sequence):
    #from ast import literal_eval
    #sequence=sequence.fillna('[]').apply(literal_eval).apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])
    counter=[]
    for line in sequence:
        counter.append(line)
        flat_list=[item for sublist in sequence for item in sublist]
        for sublist in sequence:
            for item in sublist:
                flat_list.append(item)
        counter=list(set(flat_list))
        
        return counter

In [261]:
genres=get_uniques(reves_metadata["genres"])
print("There are %d unique genres"%(len(genres)))
genres

There are 20 unique genres


['Documentary',
 'History',
 'Comedy',
 'Music',
 'Western',
 'Drama',
 'Romance',
 'TV Movie',
 'Adventure',
 'Animation',
 'Science Fiction',
 'Family',
 'Horror',
 'Crime',
 'War',
 'Thriller',
 'Action',
 'Mystery',
 'Foreign',
 'Fantasy']

In [ ]:
#we should do a barplot with the most common genres.

#### 5. Popularity

In [ ]:
reves_metadata["popularity"].describe()

In [ ]:
reves_metadata["popularity"].sample(5)

#### 6. Production Companies

In [31]:
reves_metadata["production_companies"].head(5)

0       [{'name': 'Pixar Animation Studios', 'id': 3}]
1    [{'name': 'TriStar Pictures', 'id': 559}, {'na...
3    [{'name': 'Twentieth Century Fox Film Corporat...
4    [{'name': 'Sandollar Productions', 'id': 5842}...
5    [{'name': 'Regency Enterprises', 'id': 508}, {...
Name: production_companies, dtype: object

Same as it happened with genres. Contains a list of dicts.

In [32]:
reves_metadata["production_companies"]=to_list(reves_metadata["production_companies"])

C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [264]:
reves_metadata["production_companies"].head(5)

0                            [Pixar Animation Studios]
1    [TriStar Pictures, Teitler Film, Interscope Co...
3             [Twentieth Century Fox Film Corporation]
4         [Sandollar Productions, Touchstone Pictures]
5    [Regency Enterprises, Forward Pass, Warner Bros.]
Name: production_companies, dtype: object

In [33]:
production_companies=get_uniques(reves_metadata["production_companies"])
print("There are %d unique production companies"%(len(production_companies)))
production_companies

There are 7091 unique production companies


['Echo Lake Productions',
 'Stonebridge Entertainment',
 'JEMPSA',
 'Minerva Productions',
 'Harold Greenberg Fund',
 'BigBen Films',
 '360 Pictures',
 'TMC',
 'Fuqua Films',
 'Origen Producciones Cinematograficas S.A.',
 'Hart-Sharp Entertainment',
 'Tiger Productions',
 '2T Produzione Film',
 'Canal+ Droits Audiovisuels',
 'Red Granite Pictures',
 'Morgan Creek Productions',
 'Cruel Productions',
 'Down Productions',
 'Barry & Enright Productions',
 'Pygmalion Production',
 'Bewitched',
 'Lip Sync Post',
 'Oceana Media Finance',
 'Revolver Film',
 'El Deseo',
 'Legend Production Company',
 'Programme MEDIA de la Communauté Européenne',
 'Radius-TWC',
 'Archer Gray',
 'Commission du Film Île-de-France',
 'Semtex Films',
 'World Wrestling Entertainment (WWE)',
 'Fugitive Studios',
 'Justin Pictures',
 'Lama Productions',
 'Winchester Films',
 'Ministry of Propaganda Films',
 'The Mirisch Production Company',
 'Constantin Film Produktion',
 'Polaris Film Productions & Finance',
 'Pull D

There are to many production companies for our model. Maybe we should get the most common ones, or the 20% that represent the 80% of the movies. (Around 6K movies). 

In [34]:
from collections import Counter
companies_wc=[]
for element in reves_metadata["production_companies"]:
        for company in element:
            companies_wc.append(company)
Counter(companies_wc).most_common(10)

[('Warner Bros.', 491),
 ('Universal Pictures', 463),
 ('Paramount Pictures', 395),
 ('Twentieth Century Fox Film Corporation', 341),
 ('Columbia Pictures', 236),
 ('Metro-Goldwyn-Mayer (MGM)', 207),
 ('New Line Cinema', 198),
 ('Touchstone Pictures', 158),
 ('Walt Disney Pictures', 147),
 ('Columbia Pictures Corporation', 140)]

In [44]:
production_company=[]
for element in reves_metadata["production_companies"]:
    if len(element)>0:
        first_company=element[0]
        production_company.append(first_company)
    else:
        first_company=""
        production_company.append(first_company)

In [45]:
reves_metadata["production_company"]=production_company

C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [46]:
reves_metadata["production_company"].head(5)

0                   Pixar Animation Studios
1                          TriStar Pictures
3    Twentieth Century Fox Film Corporation
4                     Sandollar Productions
5                       Regency Enterprises
Name: production_company, dtype: object

In [47]:
company_wc=[]
for element in reves_metadata["production_company"]:
        company_wc.append(element)
Counter(company_wc).most_common(6)

[('', 411),
 ('Universal Pictures', 401),
 ('Paramount Pictures', 391),
 ('Twentieth Century Fox Film Corporation', 291),
 ('Columbia Pictures', 235),
 ('New Line Cinema', 187)]

#### 7. Production Countries

In [35]:
reves_metadata["production_countries"].sample(5)

5654     [{'iso_3166_1': 'US', 'name': 'United States o...
4205     [{'iso_3166_1': 'US', 'name': 'United States o...
36893    [{'iso_3166_1': 'DE', 'name': 'Germany'}, {'is...
10398    [{'iso_3166_1': 'US', 'name': 'United States o...
15898    [{'iso_3166_1': 'US', 'name': 'United States o...
Name: production_countries, dtype: object

Almost same as before

In [36]:
reves_metadata["production_countries"]=to_list(reves_metadata["production_countries"])
production_countries=get_uniques(reves_metadata["production_countries"])
print("There are %d unique production countries"%(len(production_countries)))
production_countries[:10]

There are 98 unique production countries


C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


['Japan',
 'Slovenia',
 'Paraguay',
 'Tunisia',
 'Burkina Faso',
 'Macedonia',
 'Serbia',
 'Serbia and Montenegro',
 'Bahamas',
 'Saudi Arabia']

In [37]:
countries_wc=[]
for element in reves_metadata["production_countries"]:
        for country in element:
            countries_wc.append(country)
Counter(countries_wc).most_common(5)

[('United States of America', 5623),
 ('United Kingdom', 917),
 ('France', 576),
 ('Germany', 412),
 ('Canada', 323)]

This doesn't mean that 5 countries represent 7851 movies... We have to make sure that we make the cut leaving at least 1 country per movie. We should assume that the first country listed is the one, and take it as a reference. Then, make the countries list from that.

In [38]:
reves_metadata["production_countries"][9][0]

'United Kingdom'

In [39]:
reves_metadata["production_countries"][0][0]

'United States of America'

In [40]:
production_country=[]
for element in reves_metadata["production_countries"]:
    if len(element)>0:
        first_country=element[0]
        production_country.append(first_country)
    else:
        first_country=""
        production_country.append(first_country)

In [355]:
reves_metadata["production_countries"].sample(10)

21256                       [India]
1429     [United States of America]
4440     [United States of America]
6433     [United States of America]
5549                       [Russia]
24140    [United States of America]
43009    [United States of America]
1220     [United States of America]
24343                 [New Zealand]
10991    [United States of America]
Name: production_countries, dtype: object

In [41]:
reves_metadata["production_country"]=production_country

C:\Users\Elsa\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [42]:
reves_metadata["production_country"].head()

0    United States of America
1    United States of America
3    United States of America
4    United States of America
5    United States of America
Name: production_country, dtype: object

In [43]:
country_wc=[]
for element in reves_metadata["production_country"]:
        country_wc.append(element)
Counter(country_wc).most_common(6)

[('United States of America', 4497),
 ('United Kingdom', 548),
 ('France', 354),
 ('Canada', 267),
 ('Germany', 224),
 ('India', 204)]

#### 8. Release Date

Release Date can be a very significant variable in our model since we assume that the movies released nearby certain months or times of the year, can reach more revenue more easily. 

In [ ]:
print(reves_metadata["release_date"].sample(5))
print("There are %d null values" % (reves_metadata["release_date"].isnull().sum()))

In [ ]:
reves_metadata["release_date"][35207][8:11]

In [ ]:
type(reves_metadata["release_date"][2255])

Info is stored in string format, so we should transform that to Date format, in order to be useful. 

We can also create new columns just storing the year and the month, since the day or the exact date wont be necessary

In [ ]:
reves_metadata["release_date"]=reves_metadata["release_date"].apply(lambda x: str(x).replace("-","/"))
reves_metadata["release_date"]=reves_metadata["release_date"].apply(lambda x: pd.to_datetime(x, infer_datetime_format=True))

In [ ]:
from datetime import datetime
#for line in reves_metadata:
    #str(reves_metadata["release_date"])
    #reves_metadata["release_date"].replace("-","/")
reves_metadata["release_date"]=reves_metadata["release_date"].apply(lambda x: datetime.strptime(x, '%Y-%m-%d'))

In [ ]:
reves_metadata["release_year"]=reves_metadata["release date"][0:3]
reves_metadata["release_month"]=""
reves_metadata["release_day"]=""
def split_to_date(x):
    year=x[0:3]
    month=x[5:7]
    day=x[8:11]

#### 9. Runtime

In [ ]:
reves_metadata["runtime"].sample(5)

In [ ]:
reves_metadata["runtime"].describe()

Seems like runtime means how many days the movie has been running on cinemas. The min is 0, and that should mean that some movies have not been released. We should check if that's true, because if they have not been released, they shouldn't have revenue. 

#### 10. Title